plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,690 @@
+# Final Multigres Operator CR Definition
+
+```ascii
+[MultiCluster] 🚀 (The root CR)
+      │
+      ├── 🌍 [GlobalTopoServer] (Child CR if managed)
+      │    │
+      │    └── 🏛️ etcd Resources (if managed)
+      │
+      ├── 🤖 MultiAdmin Resources - Deployment, Services, Etc
+      │
+      ├── 💠 [MultiCell] (Child CR)
+      │    │
+      │    ├── 🚪 MultiGate Resources (Deployment, Service, etc.)
+      │    │
+      │    ├── 🧠 MultiOrch Resources (Deployment, etc.)
+      │    │
+      │    └── 📡 [LocalTopoServer] (Child CR if managed and not using global)
+      │         │
+      │         └── 🏛️ etcd Resources (if managed)
+      │
+      └── 🗃️ [MultiTableGroup] (Child CR)
+           │
+           └── 📦 [MultiShard] (Child CR)
+                │
+                └── 🏊 MultiPooler and postgres resources (pods or statefulset)
+```
+
+## MultiGres Cluster
+
+The only editable entry for the end-user. All other child CRs are read-only.
+
+```yaml
+apiVersion: multigres.com/v1alpha1
+kind: MultigresCluster
+metadata:
+  name: example-multigres-cluster
+  namespace: multigres
+spec:
+  # Images are defined globally to avoid the danger of running multiple incongruent versions at once.
+  # The operator will eventually take care of rolling updates in a safe way.
+  images:
+    multigateway: "multigres/multigres:latest"
+    multiorch: "multigres/multigres:latest"
+    multipooler: "multigres/multigres:latest"
+    multiadmin: "multigres/multigres:latest"
+    postgres: "postgres:15.3"
+
+  # ----------------------------------------------------------------
+  # Base Cluster Configuration
+  # ----------------------------------------------------------------
+  globalTopoServer:
+    rootPath: "/multigres/global"
+    managedSpec:
+      image: quay.io/coreos/etcd:v3.5.17
```

Comment:
Just so that we don't forget, CRD would need to support imagePullPolicy as well as imagePullSecrets, I believe.